packages/world/src/systems/MineSystem.sol

```diff
@@ -191,6 +164,45 @@ contract MineSystem is System {
     }
   }
 
+  function _removeRelativeBlocks(EntityId mined, ObjectTypeId minedType, Vec3 baseCoord) internal {
+    // First coord will be the base coord, the rest is relative schema coords
+    Vec3[] memory coords = minedType.getRelativeCoords(baseCoord, Orientation._get(mined));
+
+    // Only iterate through relative schema coords
+    for (uint256 i = 1; i < coords.length; i++) {
+      Vec3 relativeCoord = coords[i];
+      (EntityId relative, ObjectTypeId relativeType) = getEntityAt(relativeCoord);
+      BaseEntity._deleteRecord(relative);
+
+      _removeBlock(relative, relativeType, relativeCoord);
+    }
+  }
+
+  function _destroyEntity(
+    EntityId caller,
+    EntityId mined,
+    ObjectTypeId minedType,
+    Vec3 baseCoord,
+    bytes calldata extraData
+  ) internal {
+    // Detach program if it exists
+    ProgramId program = mined.getProgram();
+    if (program.exists()) {
+      bytes memory onDetachProgram = abi.encodeCall(IDetachProgramHook.onDetachProgram, (caller, mined, extraData));
+      program.call({ gas: SAFE_PROGRAM_GAS, hook: onDetachProgram });
```

Comment:
We dont clear the `EntityProgram` table here?

Reply:
ah good catch! We weren't doing that before either